ckanext/switzerland/helpers.py

```diff
@@ -470,3 +487,105 @@ def get_cookie_law_url():
 
 def get_cookie_law_id():
     return tk.config.get("ckanext.switzerland.cookie_law_id", "")
+
+
+def _get_datetime_from_isoformat_string(dt_string):
+    """Get a datetime object from an isoformat string, and handle empty values
+    and malformatted strings.
+
+    Returns the datetime object if possible, False if it couldn't get one.
+    """
+    if dt_string in [None, ""]:
+        return False
+    try:
+        return datetime.fromisoformat(dt_string)
+    except ValueError:
+        log.warning(f"Error getting a datetime from isoformat string: {dt_string}")
+        return False
+
+
+def convert_datetimes_for_display(dataset_or_resource_dict):
+    """Converts all datetimes in a dataset or resource to UTC, without time zone info
+    attached. CKAN's "automatic-local-datetime" HTML class and JS helper will then
+    display them in the user's local time zone.
+
+    CKAN stores all datetimes as UTC by default. For our custom datetime fields, we use
+    the server time zone (Europe/Zurich), so they have to be converted to UTC.
+    """
+    for field in CUSTOM_DATETIME_FIELDS:
+        dt = _get_datetime_from_isoformat_string(dataset_or_resource_dict.get(field))
+        if dt is False:
+            continue
+
+        dt_zh = dt.replace(tzinfo=ZURICH)
```

Comment:
What if the tzinfo is already Zurich? Do you think it is worthwhile to check first, if the data is set so we do not have to do it each time or is it likely, that we will get those dates often, even after this code has been in place for a while?